### PR TITLE
logger: allow logging HTTP traffic w/o logging everything

### PR DESCRIPTION
**Pull request by MiguelPires**
Previously, for anything to be logged SNAPD_DEBUG had to be set to a true value (1, true, etc) even if SNAPD_DEBUG_HTTP was already set. In practice, this meant that we couldn't get HTTP logs w/o getting all other logging. This changes that so we can log only HTTP traffic by setting SNAPD_DEBUG_HTTP.

### CODE REVIEW

**Review by mvo5 — APPROVED** [review]
Thank you, this looks good

**Comment by pedronis** [comment]
> This implies that "normal" snapd debug info will also be shown even if only SNAPD_DEBUG_HTTP is set but

this sounds more confusing than the status quo to me

**Comment by MiguelPires** [comment]
@pedronis thanks for the review, changed this to enable just logging HTTP traffic w/o implying or requiring SNAPD_DEBUG to be set. Changed the title and description as well

File: logger/logger.go
Comment (pedronis) [inline logger/logger.go:42]: Wondering about this name, mostly also because "DebugAlwaysf" is strange,  I don't have strong opinions but other options could be:

NoGuardDebug
DebugNotice
DebugLog

**Review by pedronis — APPROVED** [review]
thank you, some wonderings about the new Logger method name, especially when suffix with "f".

**Comment by pedronis** [comment]
@MiguelPires question: is there some docs for this that need to change?

File: logger/logger.go
Comment (mvo5) [inline logger/logger.go:41]: (nitpick^2) there is a `)` missing

**Comment by MiguelPires** [comment]
@pedronis the HACKING.md didn't specify that you needed to turn on both env vars so it's not outdated. I can add something to it so we're explicit about how it works

**Review by mvo5 — APPROVED** [review]
Thank you

**Comment by MiguelPires** [comment]
> @pedronis the HACKING.md didn't specify that you needed to turn on both env vars so it's not outdated. I can add something to it so we're explicit about how it works

Actually, I'll wait until the PR with the new env var is merged and then I'll update the HACKING to reflect both

**Pull request merged**
